### DISCUSSION TITLE
[HELP] charon.log

### DISCUSSION THREAD

**Discussion by aigenisi-aliyi** [strongswan/strongswan#1948] [General]
root:/# 
00[DMN] Starting IKE charon daemon (strongSwan 5.9.10, Linux 5.10.97+, aarch64)
00[LIB] plugin 'random': loaded successfully
00[LIB] plugin 'nonce': loaded successfully
00[LIB] plugin 'pubkey': loaded successfully
00[LIB] plugin 'counters': loaded successfully
00[LIB] plugin 'hmac': loaded successfully
00[LIB] plugin 'pem': loaded successfully
00[LIB] plugin 'openssl': loaded successfully
00[LIB] plugin 'kernel-netlink': loaded successfully
00[LIB] plugin 'socket-default': loaded successfully
00[LIB] plugin 'updown': loaded successfully
00[LIB] plugin 'vici': loaded successfully
00[KNL] getting SPD hash threshold failed: Not supported (95)
00[KNL] getting SPD hash threshold failed: Not supported (95)
00[KNL] received netlink error: Unknown device type (95)
00[KNL] failed to create XFRM interface 'xfrmi-test-1480'
00[KNL] unable to set IPSEC_POLICY on socket: Not supported (95)
00[NET] installing IKE bypass policy failed
00[KNL] unable to set IPSEC_POLICY on socket: Not supported (95)
00[NET] installing IKE bypass policy failed
00[KNL] unable to set IPSEC_POLICY on socket: Not supported (95)
00[NET] installing IKE bypass policy failed
00[KNL] unable to set IPSEC_POLICY on socket: Not supported (95)
00[NET] installing IKE bypass policy failed
00[KNL] known interfaces and IP addresses:
00[KNL]   lo
00[KNL]     127.0.0.1
00[KNL]     ::1
00[KNL]   eth_x
00[KNL]     1.1.1.3
00[KNL]     fe80::f451:b130:6420:e21a
00[KNL]   et0
00[KNL]     fe80::8317:a1b:da78:ab34
00[KNL]   rmnet2
00[KNL]     fe80::137c:a3b4:26b8:e715
00[KNL]   br0
00[KNL]     192.168.9.1
00[KNL]     fd22:9c15:283b:1a00:28a:6238:9411:b994
00[KNL]     fe80::3305:1861:2963:c036
00[KNL]   eth0.2
00[KNL]     fe80::454:91fe:dc1e:bd7b
00[KNL]   eth0.3
00[LIB] feature PUBKEY:DSA in plugin 'pem' has unmet dependency: PUBKEY:DSA
00[LIB] feature PRIVKEY:DSA in plugin 'pem' has unmet dependency: PRIVKEY:DSA
00[LIB] feature PRIVKEY:BLISS in plugin 'pem' has unmet dependency: PRIVKEY:BLISS
00[LIB] feature PUBKEY:BLISS in plugin 'pem' has unmet dependency: PUBKEY:BLISS
00[LIB] feature CERT_DECODE:PGP in plugin 'pem' has unmet dependency: CERT_DECODE:PGP
00[LIB] feature CERT_DECODE:OCSP_REQUEST in plugin 'pem' has unmet dependency: CERT_DECODE:OCSP_REQUEST
00[LIB] feature CERT_DECODE:OCSP_RESPONSE in plugin 'pem' has unmet dependency: CERT_DECODE:OCSP_RESPONSE
00[LIB] feature CERT_DECODE:X509_AC in plugin 'pem' has unmet dependency: CERT_DECODE:X509_AC
00[LIB] feature CERT_DECODE:PKCS10_REQUEST in plugin 'pem' has unmet dependency: CERT_DECODE:PKCS10_REQUEST
00[LIB] feature CONTAINER_DECODE:PKCS7 in plugin 'pem' has unmet dependency: CONTAINER_DECODE:PKCS7
00[LIB] loaded plugins: charon random nonce pubkey counters hmac pem openssl kernel-netlink socket-default updown vici
00[LIB] unable to load 10 plugin features (10 due to unmet dependencies)
00[LIB] dropped capabilities, running as uid 2423, gid 2423
00[JOB] spawning 16 worker threads
01[LIB] created thread 01 [2664]
02[LIB] created thread 02 [2665]
03[LIB] created thread 03 [2666]
05[LIB] created thread 05 [2668]
06[LIB] created thread 06 [2669]
07[LIB] created thread 07 [2670]
08[LIB] created thread 08 [2671]
09[LIB] created thread 09 [2672]
10[LIB] created thread 10 [2673]
12[LIB] created thread 12 [2675]
13[LIB] created thread 13 [2676]
14[LIB] created thread 14 [2677]
15[LIB] created thread 15 [2678]
04[LIB] created thread 04 [2667]
11[LIB] created thread 11 [2674]
16[LIB] created thread 16 [2679]
14[CFG] vici client 1 connected
03[CFG] vici client 1 requests: get-keys
07[CFG] vici client 1 requests: get-shared
13[CFG] vici client 1 requests: load-shared
13[CFG] loaded IKE shared key with id 'ike-n2n-1' for: '1.1.1.34', '1.1.1.198'
06[CFG] vici client 1 requests: get-authorities
09[CFG] vici client 1 requests: get-pools
13[CFG] vici client 1 requests: get-conns
08[CFG] vici client 1 requests: load-conn
08[CFG]  conn n2n-1:
08[CFG]   child n2n-1:
08[CFG]    rekey_time = 3600
08[CFG]    life_time = 3960
08[CFG]    rand_time = 360
08[CFG]    rekey_bytes = 0
08[CFG]    life_bytes = 0
08[CFG]    rand_bytes = 0
08[CFG]    rekey_packets = 0
08[CFG]    life_packets = 0
08[CFG]    rand_packets = 0
08[CFG]    updown = /system/bin/ipsec/_updown iptables
08[CFG]    hostaccess = 0
08[CFG]    ipcomp = 0
08[CFG]    mode = TUNNEL
08[CFG]    policies = 1
08[CFG]    policies_fwd_out = 0
08[CFG]    dpd_action = none
08[CFG]    start_action = start
08[CFG]    close_action = none
08[CFG]    reqid = 0
08[CFG]    tfc = 0
08[CFG]    priority = 0
08[CFG]    interface = (null)
08[CFG]    if_id_in = 0
08[CFG]    if_id_out = 0
08[CFG]    mark_in = 0/0
08[CFG]    mark_in_sa = 0
08[CFG]    mark_out = 0/0
08[CFG]    set_mark_in = 0/0
08[CFG]    set_mark_out = 0/0
08[CFG]    label = (null)
08[CFG]    label_mode = system
08[CFG]    inactivity = 0
08[CFG]    proposals = ESP:AES_CBC_256/HMAC_SHA2_512_256/MODP_2048/NO_EXT_SEQ, ESP:AES_CBC_256/HMAC_SHA2_384_192/MODP_2048/NO_EXT_SEQ, ESP:AES_CBC_256/HMAC_SHA2_256_128/MODP_2048/NO_EXT_SEQ, ESP:AES_CBC_192/HMAC_SHA2_512_256/MODP_2048/NO_EXT_SEQ, ESP:AES_CBC_192/HMAC_SHA2_384_192/MODP_2048/NO_EXT_SEQ, ESP:AES_CBC_192/HMAC_SHA2_256_128/MODP_2048/NO_EXT_SEQ, ESP:AES_CBC_128/HMAC_SHA2_512_256/MODP_2048/NO_EXT_SEQ, ESP:AES_CBC_128/HMAC_SHA2_384_192/MODP_2048/NO_EXT_SEQ, ESP:AES_CBC_128/HMAC_SHA2_256_128/MODP_2048/NO_EXT_SEQ
08[CFG]    local_ts = 192.168.9.0/24
08[CFG]    remote_ts = 192.168.8.0/24
08[CFG]    hw_offload = no
08[CFG]    sha256_96 = 0
08[CFG]    copy_df = 1
08[CFG]    copy_ecn = 1
08[CFG]    copy_dscp = out
08[CFG]   version = 2
08[CFG]   local_addrs = 1.1.1.34
08[CFG]   remote_addrs = 1.1.1.198
08[CFG]   local_port = 500
08[CFG]   remote_port = 500
08[CFG]   send_certreq = 1
08[CFG]   send_cert = CERT_SEND_IF_ASKED
08[CFG]   ppk_id = (null)
08[CFG]   ppk_required = 0
08[CFG]   mobike = 1
08[CFG]   aggressive = 0
08[CFG]   dscp = 0x00
08[CFG]   encap = 0
08[CFG]   dpd_delay = 10
08[CFG]   dpd_timeout = 0
08[CFG]   fragmentation = 0
08[CFG]   childless = 0
08[CFG]   unique = UNIQUE_NO
08[CFG]   keyingtries = 1
08[CFG]   reauth_time = 0
08[CFG]   rekey_time = 3600
08[CFG]   over_time = 360
08[CFG]   rand_time = 360
08[CFG]   proposals = IKE:AES_CBC_256/HMAC_SHA2_512_256/PRF_HMAC_SHA2_512/MODP_2048, IKE:AES_CBC_256/HMAC_SHA2_384_192/PRF_HMAC_SHA2_384/MODP_2048, IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048, IKE:AES_CBC_192/HMAC_SHA2_512_256/PRF_HMAC_SHA2_512/MODP_2048, IKE:AES_CBC_192/HMAC_SHA2_384_192/PRF_HMAC_SHA2_384/MODP_2048, IKE:AES_CBC_192/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048, IKE:AES_CBC_128/HMAC_SHA2_512_256/PRF_HMAC_SHA2_512/MODP_2048, IKE:AES_CBC_128/HMAC_SHA2_384_192/PRF_HMAC_SHA2_384/MODP_2048, IKE:AES_CBC_128/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
08[CFG]   if_id_in = 0
08[CFG]   if_id_out = 0
08[CFG]   local:
08[CFG]    id = 1.1.1.34
08[CFG]    class = pre-shared key
08[CFG]   remote:
08[CFG]    id = 1.1.1.198
08[CFG]    class = pre-shared key
08[CFG] added vici connection: n2n-1
08[CFG] initiating 'n2n-1'
08[MGR] checkout IKE_SA by config
08[MGR] created IKE_SA (unnamed)[1]
08[IKE] <n2n-1|1> queueing IKE_VENDOR task
08[IKE] <n2n-1|1> queueing IKE_INIT task
08[IKE] <n2n-1|1> queueing IKE_NATD task
08[IKE] <n2n-1|1> queueing IKE_CERT_PRE task
08[IKE] <n2n-1|1> queueing IKE_AUTH task
08[IKE] <n2n-1|1> queueing IKE_CERT_POST task
08[IKE] <n2n-1|1> queueing IKE_CONFIG task
08[IKE] <n2n-1|1> queueing IKE_AUTH_LIFETIME task
08[IKE] <n2n-1|1> queueing IKE_MOBIKE task
08[IKE] <n2n-1|1> queueing IKE_ESTABLISH task
08[IKE] <n2n-1|1> queueing CHILD_CREATE task
08[IKE] <n2n-1|1> activating new tasks
08[IKE] <n2n-1|1>   activating IKE_VENDOR task
08[IKE] <n2n-1|1>   activating IKE_INIT task
08[IKE] <n2n-1|1>   activating IKE_NATD task
08[IKE] <n2n-1|1>   activating IKE_CERT_PRE task
08[IKE] <n2n-1|1>   activating IKE_AUTH task
08[IKE] <n2n-1|1>   activating IKE_CERT_POST task
08[IKE] <n2n-1|1>   activating IKE_CONFIG task
08[IKE] <n2n-1|1>   activating IKE_AUTH_LIFETIME task
08[IKE] <n2n-1|1>   activating IKE_MOBIKE task
08[IKE] <n2n-1|1>   activating IKE_ESTABLISH task
08[IKE] <n2n-1|1>   activating CHILD_CREATE task
08[IKE] <n2n-1|1> initiating IKE_SA n2n-1[1] to 1.1.1.198
08[IKE] <n2n-1|1> IKE_SA n2n-1[1] state change: CREATED => CONNECTING
08[CFG] <n2n-1|1> configured proposals: IKE:AES_CBC_256/HMAC_SHA2_512_256/PRF_HMAC_SHA2_512/MODP_2048, IKE:AES_CBC_256/HMAC_SHA2_384_192/PRF_HMAC_SHA2_384/MODP_2048, IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048, IKE:AES_CBC_192/HMAC_SHA2_512_256/PRF_HMAC_SHA2_512/MODP_2048, IKE:AES_CBC_192/HMAC_SHA2_384_192/PRF_HMAC_SHA2_384/MODP_2048, IKE:AES_CBC_192/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048, IKE:AES_CBC_128/HMAC_SHA2_512_256/PRF_HMAC_SHA2_512/MODP_2048, IKE:AES_CBC_128/HMAC_SHA2_384_192/PRF_HMAC_SHA2_384/MODP_2048, IKE:AES_CBC_128/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
08[CFG] <n2n-1|1> sending supported signature hash algorithms: sha256 sha384 sha512 identity
08[ENC] <n2n-1|1> generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(HASH_ALG) N(REDIR_SUP) ]
08[NET] <n2n-1|1> sending packet: from 1.1.1.34[500] to 1.1.1.198[500] (808 bytes)
08[MGR] <n2n-1|1> checkin IKEv2 SA n2n-1[1] with SPIs 52d2e7accd5d109d_i 0000000000000000_r
08[MGR] <n2n-1|1> checkin of IKE_SA successful
13[CFG] vici client 1 disconnected
03[MGR] checkout IKEv2 SA by message with SPIs 52d2e7accd5d109d_i ddf6046ac9e87c2c_r
03[MGR] IKE_SA n2n-1[1] successfully checked out
03[NET] <n2n-1|1> received packet: from 1.1.1.198[500] to 1.1.1.34[500] (464 bytes)
03[ENC] <n2n-1|1> parsed IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(HASH_ALG) N(CHDLESS_SUP) N(MULT_AUTH) ]
03[IKE] <n2n-1|1> received SIGNATURE_HASH_ALGORITHMS notify
03[IKE] <n2n-1|1> received CHILDLESS_IKEV2_SUPPORTED notify
03[CFG] <n2n-1|1> selecting proposal:
03[CFG] <n2n-1|1>   proposal matches
03[CFG] <n2n-1|1> received proposals: IKE:AES_CBC_256/HMAC_SHA2_512_256/PRF_HMAC_SHA2_512/MODP_2048
03[CFG] <n2n-1|1> configured proposals: IKE:AES_CBC_256/HMAC_SHA2_512_256/PRF_HMAC_SHA2_512/MODP_2048, IKE:AES_CBC_256/HMAC_SHA2_384_192/PRF_HMAC_SHA2_384/MODP_2048, IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048, IKE:AES_CBC_192/HMAC_SHA2_512_256/PRF_HMAC_SHA2_512/MODP_2048, IKE:AES_CBC_192/HMAC_SHA2_384_192/PRF_HMAC_SHA2_384/MODP_2048, IKE:AES_CBC_192/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048, IKE:AES_CBC_128/HMAC_SHA2_512_256/PRF_HMAC_SHA2_512/MODP_2048, IKE:AES_CBC_128/HMAC_SHA2_384_192/PRF_HMAC_SHA2_384/MODP_2048, IKE:AES_CBC_128/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
03[CFG] <n2n-1|1> selected proposal: IKE:AES_CBC_256/HMAC_SHA2_512_256/PRF_HMAC_SHA2_512/MODP_2048
03[CFG] <n2n-1|1> received supported signature hash algorithms: sha256 sha384 sha512 identity
03[IKE] <n2n-1|1> reinitiating already active tasks
03[IKE] <n2n-1|1>   IKE_CERT_PRE task
03[IKE] <n2n-1|1>   IKE_AUTH task
03[IKE] <n2n-1|1> authentication of '1.1.1.34' (myself) with pre-shared key
03[IKE] <n2n-1|1> successfully created shared key MAC
03[CFG] <n2n-1|1> proposing traffic selectors for us:
03[CFG] <n2n-1|1>  192.168.9.0/24
03[CFG] <n2n-1|1> proposing traffic selectors for other:
03[CFG] <n2n-1|1>  192.168.8.0/24
03[CFG] <n2n-1|1> configured proposals: ESP:AES_CBC_256/HMAC_SHA2_512_256/NO_EXT_SEQ, ESP:AES_CBC_256/HMAC_SHA2_384_192/NO_EXT_SEQ, ESP:AES_CBC_256/HMAC_SHA2_256_128/NO_EXT_SEQ, ESP:AES_CBC_192/HMAC_SHA2_512_256/NO_EXT_SEQ, ESP:AES_CBC_192/HMAC_SHA2_384_192/NO_EXT_SEQ, ESP:AES_CBC_192/HMAC_SHA2_256_128/NO_EXT_SEQ, ESP:AES_CBC_128/HMAC_SHA2_512_256/NO_EXT_SEQ, ESP:AES_CBC_128/HMAC_SHA2_384_192/NO_EXT_SEQ, ESP:AES_CBC_128/HMAC_SHA2_256_128/NO_EXT_SEQ
03[IKE] <n2n-1|1> establishing CHILD_SA n2n-1{1}
03[KNL] <n2n-1|1> allocating SPI failed: Not supported (95)
03[KNL] <n2n-1|1> unable to get SPI
03[IKE] <n2n-1|1> unable to allocate SPIs from kernel
03[CHD] <n2n-1|1> CHILD_SA n2n-1{1} state change: CREATED => DESTROYING
03[MGR] <n2n-1|1> checkin and destroy IKE_SA n2n-1[1]
03[IKE] <n2n-1|1> IKE_SA n2n-1[1] state change: CONNECTING => DESTROYING
03[MGR] checkin and destroy of IKE_SA successful
16[MGR] checkout IKEv2 SA with SPIs 52d2e7accd5d109d_i 0000000000000000_r
16[MGR] IKE_SA checkout not successful

**Comment by aigenisi-aliyi** [comment]
[help] this connect is failed

**Reply by strongX509** [reply]
Somehow your Linux 50.10.97 aarch64 kernel does not support IPsec so that all communication via the XFRM kernel interface fails.
